### DISCUSSION TITLE
Extending the GitHub's Web User Interface

### DISCUSSION THREAD

**Discussion by gbidsilva** [community/community#23742] [New to GitHub]
<p>Is there a way that we can extend GitHub’s web user interface functionality by adding new UI components such as buttons, labels and drop-downs?</p>
<p>The idea behind this is, we need to showcase some other 3rd party application data such as web links within the Github repository (or other) pages.</p>
<p>Is this kind of plugin development possible with GitHub architecture?</p>
<p>All I could find is details on GitHub Apps in which we can give permission to outside hosted application but nothing like extending the existing UI functionality.</p>
<p>Thank You.</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi there! :wave: Welcome to the Community!</p>
<p>The kind of extension you’re asking for isn’t offered natively in GitHub, but it is absolutely possible to create a browser extension that will do this for you. There are quite a few projects out there that supplement GitHub’s default behavior.</p>
<p>Probably the most well known is this one:</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/sindresorhus/refined-github" target="_blank">GitHub</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/19803303/181079169-cdc5883a-c8eb-405b-a65d-1d672a4ad795.jpeg" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/f72a2629ff23902f32578499431ce1a257cd320e_2_690x345.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/f/f72a2629ff23902f32578499431ce1a257cd320e_2_1035x517.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/f/f72a2629ff23902f32578499431ce1a257cd320e_2_1380x690.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f72a2629ff23902f32578499431ce1a257cd320e_2_10x10.png"></div>

<h3><a href="https://github.com/sindresorhus/refined-github" target="_blank">sindresorhus/refined-github</a></h3>

<p>:octocat: Browser extension that simplifies the GitHub interface and adds useful features - sindresorhus/refined-github</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
